### PR TITLE
Feature/component sub generator

### PR DESCRIPTION
**Pull request by zhawkins**
- Moves "starter-kit" components to their own subgenerator.
- Updates the component subgenerator so it does more and works with pattern lab.
- Adds a "generate" npm script for quickly adding additional components.
- Adds a default field template that's cleaner.
- Fixes eslint bug with 'use strict' behaving incorrectly for gulp / node files vs Drupal js files.

### CODE REVIEW

File: generators/app/templates/_src/patterns/global/global.scss [hunk 1]
```diff
@@ -44,34 +36,41 @@ html {
   }
 }
 
-// Sections.
+html {
+  @include font-stack-primary;
+  font-size: 100%;
```
Comment (erichuffman) [inline generators/app/templates/_src/patterns/global/global.scss]: fwiw, I like to see the `html` selector specify 62.5% font-size, and then the `body` selector specify 1.6rem font-size for a baseline font size of 16px. That makes it easier to specify other font sizes in rems later on (2.4rem = 24px, etc.)

Comment (zhawkins) [reply generators/app/templates/_src/patterns/global/global.scss]: I haven't done that in a while but I don't mind it. I'll get some feels on it in the FED channel.

File: generators/starter-kit/index.js [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+          fs.readFileSync(
+            path.resolve(this.destinationPath('package.json')), 'utf8'
+          )
+        );
+        defaultThemeName = pkg.name;
+      }
+      catch (err) {
+        assert.fail(
+          `
+🚨 ${chalk.red(this.destinationPath('package.json'))} ${chalk.red('is missing')}.
```
[the remark this answers is not in the document]
Comment (zhawkins) [reply generators/starter-kit/index.js]: It is! 🚨

File: generators/app/templates/_src/patterns/global/global.scss (hunk 1 above)
Comment (zhawkins) [reply generators/app/templates/_src/patterns/global/global.scss]: After talking to others and thinking on this more, I've decided to leave it as is. That feels like the sort of thing I want to leave up to the project lead and not something I want to recommend for everyone. Thanks though 👍

**Pull request merged**